packages/prerender-fargate/lib/prerender/server.js

```diff
@@ -36,8 +36,114 @@ server.use({
+                    } else {
+                        console.error(err);
+                    }
+
+                    next();
+                });
+            },
+
+            pageLoaded: function(req, res, next) {
+                const statusCodesToCache = ['200', '301', '302'];
+                let metaTagStatusCode = 200;
+                let location = '';
+
+                if (req.prerender.content && req.prerender.renderType == 'html') {
+                    let statusMatch = /<meta[^<>]*(?:name=['"]prerender-status-code['"][^<>]*content=['"]([0-9]{3})['"]|content=['"]([0-9]{3})['"][^<>]*name=['"]prerender-status-code['"])[^<>]*>/i,
+                    headerMatch = /<meta[^<>]*(?:name=['"]prerender-header['"][^<>]*content=['"]([^'"]*?): ?([^'"]*?)['"]|content=['"]([^'"]*?): ?([^'"]*?)['"][^<>]*name=['"]prerender-header['"])[^<>]*>/gi,
+                    head = req.prerender.content.toString().split('</head>', 1).pop(),
+                    // statusCode = 200,
+                    match;
+
+                    if (match = statusMatch.exec(head)) {
+                        metaTagStatusCode = match[1] || match[2];
+                        req.prerender.content = req.prerender.content.toString().replace(match[0], '');
+                        console.log("metaTagStatusCode: " + metaTagStatusCode);
+                    }
+
+                    while (match = headerMatch.exec(head)) {
+                        location = he.decode(match[2] || match[4]);
+                        res.setHeader(match[1] || match[3], location);
+                        req.prerender.content = req.prerender.content.toString().replace(match[0], '');
+                    }
+
+                    // Skip caching for the http response codes not in the list, such as 404
+                    if ( ! statusCodesToCache.includes(metaTagStatusCode.toString()) ) {
+                        console.log("metaTagStatusCode " + metaTagStatusCode + " is not in the cachable code list. Returning without caching.");
+                        return res.send(metaTagStatusCode, req.prerender.content);
+                    }
+                }
+
+                if(req.prerender.statusCode !== 200) {
+                    return next();
+                }
+
+                // Override req.prerender.statusCode with the StatusCode returned via the meta tag.
+                // If metaTagStatusCode is not in the statusCodesToCache array or req.prerender.statusCode is not 200, then this line wouldn't be reached. Therefore no if condition for this overriding is needed.
+                req.prerender.statusCode = metaTagStatusCode;
+                console.log("Caching the object with statusCode " + req.prerender.statusCode);
+
+                var key = req.prerender.url;
+                var s3Metadata = {
+                    httpreturncode: req.prerender.statusCode.toString()
+                }
+
+                if (location) {
```

Comment:
I feel like this could be done earlier on, it feels a bit out of place referencing the location variable here. I reckon you could change this so `location` is scoped inside an existing `if` statement above rather than being scoped to the function